Fix test_config_vault.py: 2 failures on default/env vault URL assertions

## What fails

- `test_vault_configuration_defaults`: `AssertionError: assert None == 'dev-only-token'`
- `test_vault_configuration_from_environment`: `AssertionError: assert 'http://vault:8200' == 'http://env-vault:8200'`

## Root cause

The `Settings` class in `trading-bot/app/config.py` is no longer returning the expected default Vault token or honoring the `VAULT_URL` env var. Likely `pytest.ini` `env =` block leakage or the test not clearing env vars before asserting defaults.

`pytest.ini` defines `VAULT_URL` nowhere, but the test asserts `vault_url == 'http://env-vault:8200'` after an env override — the override isn't taking effect, meaning either monkeypatch scope is wrong or Settings is cached.

## How to reproduce
```bash
pytest tests/unit/test_config_vault.py -x
```

## Fix
1. Use `monkeypatch.setenv` and re-instantiate `Settings` inside the test (not at module scope).
2. Confirm the default token value matches the one in `trading-bot/app/config.py` (likely renamed from `dev-only-token` to `None` when Vault integration became explicit).


---
*Detected on CI run for PR #178 (https://github.com/AIgen-Solutions-s-r-l/SpreadPilot/actions/runs/24424919660). These failures are also present on `main` (last 3 runs of `CI Pipeline` are `failure`). They block the CI gate for every PR.*

--- a/tests/unit/test_config_vault.py
+++ b/tests/unit/test_config_vault.py
@@ -27,12 +27,20 @@ class TestSettingsVaultIntegration:
             vault_enabled=True,
         )
 
-    def test_vault_configuration_defaults(self):
-        """Test Vault configuration default values."""
+    def test_vault_configuration_defaults(self, monkeypatch):
+        """Test Vault configuration default values.
+
+        vault_token has no default — the startup validation introduced in #84
+        requires it to be set explicitly via env var or Vault bootstrap.
+        """
+        # Strip any leaked env vars from pytest.ini or the host shell
+        for var in ("VAULT_ADDR", "VAULT_TOKEN", "VAULT_MOUNT_POINT", "VAULT_ENABLED"):
+            monkeypatch.delenv(var, raising=False)
+
         settings = Settings()
 
         assert settings.vault_url == "http://vault:8200"
-        assert settings.vault_token == "dev-only-token"
+        assert settings.vault_token is None
         assert settings.vault_mount_point == "secret"
         assert settings.vault_enabled is True
 

--- a/tests/unit/test_config_vault.py
+++ b/tests/unit/test_config_vault.py
@@ -156,6 +156,21 @@ class TestSettingsVaultIntegration:
         assert mock_vault_client.vault_token == "custom-token"
         assert mock_vault_client.mount_point == "custom-secret"
 
+    def test_project_id_reads_google_cloud_project_env_var(self, monkeypatch):
+        """Regression guard for the `project_id` env-var bug fixed alongside #181.
+
+        `project_id` uses `validation_alias=AliasChoices('project_id',
+        'GOOGLE_CLOUD_PROJECT')` because pydantic-settings v2 cannot auto-map
+        the attribute name to the (non-matching) env var GOOGLE_CLOUD_PROJECT.
+        """
+        monkeypatch.setenv("GOOGLE_CLOUD_PROJECT", "prod-gcp-project-abc")
+        # Clear Vault env vars so they don't trigger validation elsewhere
+        for var in ("VAULT_ADDR", "VAULT_TOKEN", "VAULT_MOUNT_POINT", "VAULT_ENABLED"):
+            monkeypatch.delenv(var, raising=False)
+
+        settings = Settings()
+        assert settings.project_id == "prod-gcp-project-abc"
+
     def test_trading_mode_validation_unchanged(self):
         """Test that trading mode validation still works with Vault integration."""
         # Test valid trading mode

--- a/trading-bot/app/config.py
+++ b/trading-bot/app/config.py
@@ -2,7 +2,7 @@
 
 from functools import lru_cache
 
-from pydantic import Field, validator
+from pydantic import AliasChoices, Field, validator
 from pydantic_settings import BaseSettings
 from spreadpilot_core.logging import get_logger
 from spreadpilot_core.utils.vault import get_vault_client
@@ -153,9 +153,11 @@ class Settings(BaseSettings):
     )
 
     # Vault configuration
+    # AliasChoices accepts both the Python field name (for test construction
+    # via Settings(vault_url=...)) and the standard Vault env var VAULT_ADDR.
     vault_url: str = Field(
         default="http://vault:8200",
-        env="VAULT_ADDR",
+        validation_alias=AliasChoices("vault_url", "VAULT_ADDR"),
         description="HashiCorp Vault server URL",
     )
     vault_token: str | None = Field(

--- a/trading-bot/app/config.py
+++ b/trading-bot/app/config.py
@@ -17,9 +17,12 @@ class Settings(BaseSettings):
     """
 
     # Google Cloud Project
+    # Same AliasChoices pattern as vault_url: the standard env var is
+    # GOOGLE_CLOUD_PROJECT but the Python attribute is project_id, so
+    # pydantic-settings v2 auto-mapping cannot bridge them.
     project_id: str = Field(
         default="spreadpilot-dev",
-        env="GOOGLE_CLOUD_PROJECT",
+        validation_alias=AliasChoices("project_id", "GOOGLE_CLOUD_PROJECT"),
         description="Google Cloud Project ID",
     )
 

